app/javascript/dashboard/mixins/agentMixin.js

```diff
@@ -0,0 +1,35 @@
+import { mapGetters } from 'vuex';
+
+export default {
+  computed: {
+    getAgents() {
+      return this.$store.getters['inboxAssignableAgents/getAssignableAgents'](
+        this.inboxId
+      );
+    },
+    ...mapGetters({
+      currentUser: 'getCurrentUser',
+    }),
+    agentsList() {
+      const agents = this.getAgents || [];
+      return [
+        {
+          confirmed: true,
+          name: 'None',
+          id: 0,
+          role: 'agent',
+          account_id: 0,
+          email: 'None',
+        },
+        ...agents,
+      ].map(item =>
+        item.id === this.currentUser.id
+          ? {
+              ...item,
+              availability_status: this.currentUser.availability_status,
+            }
+          : item
```

Comment:
Is it better to update the state when the action is performed instead of merging it here? Are there any other conditions which I have missed?

Reply:
Mainly this action is performing after the user changes the status from the account menu. To update the assignable agent's list, we need to pass the currently selected inbox id. Unfortunately, we are not keeping the `inboxId` in auth module. That's why I moved this logic to agent mixin.

Reply:
Sounds good.